lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvConnection.php

```diff
@@ -155,7 +170,7 @@ public function lastInsertId($name = null)
             $stmt = $this->query('SELECT @@IDENTITY');
         }
 
-        return $stmt->fetchColumn();
+        return $stmt->fetchColumn() ?: '0';
```

Comment:
I don't like`?:` here, it hides possible _error values_ (would that be `false`, `'0'` or `0` here?) - is it possible to use `!== false` for example?

Reply:
In the original proposal, `lastInsertId()` suppresses errors everywhere by design (see the OCI implementation specifically). Not sure I understand and agree with it. We definitely could improve here.